src/main/java/org/commcare/formplayer/session/FormSession.java

```diff
@@ -145,13 +151,16 @@ public FormSession(UserSqlSandbox sandbox,
             @Nullable SessionFrame sessionFrame,
             RemoteInstanceFetcher instanceFetcher) throws Exception {
 
-        this.formDef = formDef;
-        session = new SerializableFormSession(
+        // use this.formDef to mutate (e.g., inject instance content, set callout handler)
+        this.formDef = FormDefStringSerializer.deserialize(serializableFormDefinition.getSerializedFormDef());
```

Comment:
It looks like we take a `FormDef`, convert it to a `serializableFormDefinition`, pass that to this constructor, the immediately derserialize it to get a `FormDef` back. Is that not expensive?

Reply:
Yeah good point. Would it be odd to add a non-postgres property to SerializableFormDefinition to hold the deserializedFormDef? I think I was hesitant to do that originally and then didn't come back to optimizing this part.

Reply:
Ok I think that isn't allowed since hibernate isn't able to determine what column to map that property to, unless there is a way to omit properties from hibernate that I don't know of. 

Alternatively, pushing the form definition service/creation of a serializable form definition into the form session seems like the next best solution. This would require passing a few additional args into the form session constructor which is already bloated, but seems worth it.

Reply:
You should be able to use the [@Transient](https://docs.oracle.com/javaee/7/api/javax/persistence/Transient.html) annotation to prevent hibernate from mapping the field.

Reply:
That's perfect. Thanks for highlighting Simon. See 3d32f0ddb7ea2538e7cfedbeef8dae423c0fba3a